pkg/controller/admissionchecks/provisioning/controller.go

```diff
@@ -247,16 +216,18 @@ func (c *Controller) syncOwnedProvisionRequest(ctx context.Context, wl *kueue.Wo
 		attempt := int32(1)
 		shouldCreatePr := false
 		if exists {
-			if isFailed(oldPr) || (isBookingExpired(oldPr) && !workload.IsAdmitted(wl)) {
+			if (isFailed(oldPr) || (isBookingExpired(oldPr) && !workload.IsAdmitted(wl))) &&
 				// if the workload is Admitted we don't want to retry on BookingExpired
+				ac != nil && ac.State == kueue.CheckStatePending {
+				// if the workload is in Retry/Rejected state we don't create another ProvReq
 				attempt = getAttempt(log, oldPr, wl.Name, checkName)
-				if attempt <= c.maxRetries {
-					remainingTime := c.remainingTimeToRetry(oldPr, attempt)
-					if remainingTime <= 0 {
+				if attempt <= ptr.Deref(retryStrategy.BackoffLimitCount, kueue.DefaultBackoffLimitCount) {
+					remainingTimeToRetry := c.remainingTimeToRetry(wlInfo, oldPr, attempt, prc)
+					if remainingTimeToRetry <= 0 {
 						shouldCreatePr = true
 						attempt += 1
-					} else if requeAfter == nil || remainingTime < *requeAfter {
```

Comment:
Why is this removed? I think the code makes sense to use minimum as the  effective `requeueAfter` rather than the last, but I agree we may not have tests for it

Reply:
Done